### ISSUE TITLE
add recombination map(s) for BosTau

### ISSUE THREAD

**Issue by grahamgower** [popsim-consortium/stdpopsim#602]
**Short description of what you'd like added:**
The recently added BosTau species could be improved by the addition of one or more recombination maps.

**Rationale and considerations:**
Ma et al. 2015 (https://doi.org/10.1371/journal.pgen.1005387) have recombination map data for Holstein cattle available on dryad (https://datadryad.org/stash/dataset/doi:10.5061/dryad.q2q84).
Additional cattle breed recombination maps are available as part of Shen et al. 2018 (https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4705-y#Sec16)

**Comment by gregorgorjanc** [comment]
@grahamgower Do you want to assign this to me?

**Comment by grahamgower** [comment]
Thanks @gregorgorjanc! I've just realised that you'll need to do a liftover here as well (see #574 and #571).

**Comment by gregorgorjanc** [comment]
@grahamgower I have now pulled recombination data from dryad and started to look at this leftover. The paper (https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005387) that inferred recombination map says

```To study recombination, we included up to 59,309 genome-wide SNPs after quality control filtering and used the USDA-AGIL SNP coordinates that showed a higher quality than the UMD3.1 assembly [30,33]. Previously, several SNPs were relocated from the UMD3.1 assembly to the USDA-AIGL coordinates in cooperation with researchers from the University of Missouri (R. D. Schnabel), and the University of Guelph (M. Sargolzaei and J. Johnston).```

While the stdpopsim BosTau uses assembly ARS-UCD1.2 (https://github.com/popsim-consortium/stdpopsim/blob/main/stdpopsim/catalog/BosTau/genome_data.py).

So the purpose of leftover is to convert the recombination map from the "USDA-AGIL SNP coordinates" to the new assembly (ARS-UCD1.2)?

**Comment by jradrion** [comment]
Hmm, IIRC to use the maintenance code "as-is" you would first need to add the map with the original coordinates and also add the older assembly too. Then when you run it you would set `--map` to whatever you named that original map in stdpopsim.

**Comment by jradrion** [comment]
It's not set up to do the liftover outside of `stdpopsim`, although with a little modification that could be done too.

**Comment by gregorgorjanc** [comment]
Thanks @jradrion!

Do you expect I will have issues with the liftover because the developed recombination map used semi-custom SNP coordinates (an improved UMD3.1/BosTau8), but the chain file is between the official UMD3.1/BosTau8 and ARS-UCD1.2/BosTau9 assemblies?

**Comment by jradrion** [comment]
> Do you expect I will have issues with the liftover because the developed recombination map used semi-custom SNP coordinates (an improved UMD3.1/BosTau8), but the chain file is between the official UMD3.1/BosTau8 and ARS-UCD1.2/BosTau9 assemblies?

Yeah, that could be problematic. The chain file needs to be in the same coordinate space as the original map.

**Comment by jradrion** [comment]
Do you know if the coordinates are actually different and not just some of the annotations?  I don't think it's going to work if the coordinates differ.

**Comment by grahamgower** [comment]
@gregorgorjanc I'm a little fuzzy on what the USDA-AGIL SNP coordinates are supposed to be. Is this a separate assembly? Do you have a url/reference for it?

**Comment by gregorgorjanc** [comment]
@grahamgower it was an "USDA in-house" map that was improved compared to UMD3.1/BosTau8 assembly. I need to talk to the authors - maybe they have remapped to the recent assembly. Will report back.

**Comment by gregorgorjanc** [comment]
I looked into this today, as we would like to have this data for one of our projects. While I managed to port the recombination rate from [this publication ](https://doi.org/10.1371/journal.pgen.1005387) to the recent assembly (using my own script and information from https://www.animalgenome.org/repository/cattle/UMC_bovine_coordinates), I am losing some markers (not placed on the assembly or changed chromosome), and the total recombination map length is reduced by ~1.5 Morgans, which I don't think is what we want.

This made me think how does lifover of a recombination map from one to another assembly work! Isn't this an impossible task? Recombination maps report the recombination rate between two neighbouring markers. If I change slightly the position of markers, the marker order might not change, and only marker positions change a little bit, so all is well. However, if marker positions change a lot, and hence also their order, then there is no way to lift over recombination rates.

Any suggestions what to do here @jradrion @andrewkern?

One alternative I see is to create a simpler recombination map where I somehow interpolate over the removed markers and incorporate their past recombination values or something like this ... Any suggestions welcome.

**Comment by petrelharp** [comment]
The only option I see is to simply remove those markers whose position changes a lot: so, I think this would work like the following: suppose that in the original assembly the distance between markers `a` and `b` is `x`cM; while between `b` and `c` it is `y`cM. Then suppose that `b` gets mapped somewhere else entirely; remove it, and set the distance between `a` and `c` to `x+y`cM.

Otherwise, it seems like you just need to re-infer a whole new map, and then publish this, somehow?

**Comment by gregorgorjanc** [comment]
OK, I have now done this. 

I loose about ~5K markers out of ~59K in "lifting over".

For some chromosomes this works well (see [chr1](https://github.com/gregorgorjanc/stdpopsim_cattle_rec_map/blob/main/Ma_et_al_2015/chr1.png)), but some parts of the assembly have changed quite a bit (see [chr28](https://github.com/gregorgorjanc/stdpopsim_cattle_rec_map/blob/main/Ma_et_al_2015/chr28.png) and I don't yet know how to easily handle that. I contacted the author ...

I found some other maps that are against the new assembly - see https://github.com/gregorgorjanc/stdpopsim_cattle_rec_map

This process shows that [lifting over recombination rate maps](https://github.com/popsim-consortium/stdpopsim/issues/571) is not straightforward. Maybe this should be noted somewhere in the docs https://popsim-consortium.github.io/stdpopsim-docs/stable/development.html#lifting-over-a-recombination-genetic-map?

**Comment by gregorgorjanc** [comment]
@andrewkern I have sorted one genetic map for cattle (with several in the pipeline for a later date - see https://github.com/gregorgorjanc/stdpopsim_cattle_rec_map). The `tar.gz` is at https://github.com/gregorgorjanc/stdpopsim_cattle_rec_map/blob/main/Brekke_et_al_2023/Brekke2023_NRF_ARS-UCD1.2.tar.gz and once you give me `url` and `sha256` I can update my WIP at https://github.com/gregorgorjanc/stdpopsim/tree/BosTauBrekke2023GeneticMap

I followed the guidelines at https://popsim-consortium.github.io/stdpopsim-docs/stable/development.html#adding-a-recombination-genetic-map-or-annotation and HapMap file details at https://tskit.dev/msprime/docs/stable/api.html#msprime.RateMap.read_hapmap, but would appreciate checks from your end.

Thanks!
